homeassistant/components/tolosauna/__init__.py

```diff
@@ -0,0 +1,113 @@
+"""Component to control TOLO Sauna/Steam Bath."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any, cast
+
+from tololib import ToloClient
+from tololib.errors import ResponseTimedOutError
+from tololib.message_info import SettingsInfo, StatusInfo
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from ...helpers.entity import DeviceInfo
+from .const import DEFAULT_RETRY_COUNT, DEFAULT_RETRY_TIMEOUT, DOMAIN
+
+PLATFORMS = ["climate"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up tolosauna from a config entry."""
+    client = ToloClient(entry.data[CONF_HOST])
+    coordinator = ToloSaunaUpdateCoordinator(hass, entry)
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {"client": client, "coordinator": coordinator}
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class ToloSaunaUpdateCoordinator(DataUpdateCoordinator):
+    """DataUpdateCoordinator for TOLO Sauna."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize ToloSaunaUpdateCoordinator."""
+        self._config_entry = entry
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=f"{entry.title} ({entry.data[CONF_HOST]}) Data Update Coordinator",
+            update_method=self._update_data,
+            update_interval=timedelta(seconds=3),
+        )
+
+    async def _update_data(self) -> dict[str, Any]:
```

Comment:
If you rename this method to `_async_update_data` you could omit `update_method=self._update_data,` from the `super().__init__`.

Reply:
Done.